staging/src/k8s.io/apiserver/pkg/endpoints/handlers/negotiation/negotiate_test.go

```diff
@@ -64,6 +65,93 @@ func (n *fakeNegotiater) DecoderToVersion(serializer runtime.Decoder, gv runtime
 
 var fakeCodec = runtime.NewCodec(runtime.NoopEncoder{}, runtime.NoopDecoder{})
 
+var simpleTestEndpointRestrictions = testEndpointRestrictions{}
+
+type testEndpointRestrictions struct{}
+
+func (testEndpointRestrictions) AllowsConversion(schema.GroupVersionKind) bool { return true }
+func (testEndpointRestrictions) AllowsServerVersion(string) bool               { return true }
+func (testEndpointRestrictions) AllowsStreamSchema(s string) bool              { return true }
+
+func TestNegotiateOutputMediaType(t *testing.T) {
+	testCases := []struct {
+		accept      string
+		req         *http.Request
+		ns          *fakeNegotiater
+		serializer  runtime.Serializer
+		contentType string
+		convert     schema.GroupVersionKind
+		params      map[string]string
+		errFn       func(error) bool
+	}{
+		{
+			req: &http.Request{
+				Header: http.Header{
+					"Accept": []string{"text/plain;as=Test;v=v12;g=meta.k8s.io, application/json"},
+				},
+			},
+			contentType: "text/plain",
+			convert:     schema.GroupVersionKind{Group: "meta.k8s.io", Kind: "Test", Version: "v12"},
+			ns:          &fakeNegotiater{serializer: fakeCodec, types: []string{"application/json", "text/plain"}},
+			serializer:  fakeCodec,
+		},
+		{
+			req: &http.Request{
+				Header: http.Header{
+					"Accept": []string{"application/json;as=PartialObjectMetadataList;v=v1alpha1;g=meta.k8s.io, application/json"},
+				},
+			},
+			contentType: "application/json",
+			convert:     schema.GroupVersionKind{Group: "meta.k8s.io", Kind: "PartialObjectMetadataList", Version: "v1alpha1"},
+			ns:          &fakeNegotiater{serializer: fakeCodec, types: []string{"application/json", "text/plain"}},
+			serializer:  fakeCodec,
+		},
+		{
+			req: &http.Request{
+				Header: http.Header{
+					"Accept": []string{"application/json;as=PartialObjectMetadataList;v=v1alpha1;g=meta.k8s.io"},
+				},
+			},
+			contentType: "application/json",
```

Comment:
why would this not return "application/json;as=PartialObjectMetadataList;v=v1alpha1;g=meta.k8s.io"?